### ISSUE TITLE
Test Code: After Step Event, Upload All Label Data for that Pano ID

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#947]
Because of the problems we have had in losing data, I was thinking we could do this: after the user registers a step event on a panorama (that is, is about to leave said panorama), as test code, we should re-upload all labels and associated metadata for that panoid and check to make sure we have the same exact data on our backend.

We could then comment this code out once we are confident our fixes work (or just put it in a If DEBUG==True conditional).

**Comment by jonfroehlich** [comment]
@sbower213, can you write an update on this before you go on vacation please?

**Comment by sbower213** [comment]
No updates on this issue so far. I started working on it but quickly went back to fixing up my code in #945 to get that ready for testing and merging. Once that code is added to production, it will be much easier to work on this branch, since we need the changes from #937 to be incorporated first.

**Comment by jonfroehlich** [comment]
Got it. Thanks.

**Comment by jonfroehlich** [comment]
@sbower213 I know you are no longer contributing code to this project but could you update this Issue with a breakdown of where we stand on this so someone could come in and possibly finish it up?

**Comment by jonfroehlich** [comment]
Are we confident that this is no longer an issue @misaugstad? It seems like we should have some test cases setup to ensure that we are not losing any data...

**Comment by misaugstad** [comment]
Ahh yes I had figured that we were confident that we were getting all the data after the close scrutiny of issue #1648 (where we were losing data on page unload). Though upon closer reading, I see that this quite the same thing, and this seems important enough to possibly want dedicated tests.

Do you remember what we are talking about here with losing data?

**Comment by jonfroehlich** [comment]
Yes, I do. This was a critical data loss issue where we were only uploading data after every X events after pano load. This was during your first summer with us. We solved this problem (I believe) but never had a test infrastructure setup to fully validate our solutions.

**Comment by misaugstad** [comment]
Yes, we now send data back to the server after a label has been added, and every 200 logged interactions (which is quite often given the logging of mouse movement). I think that it's appropriate to close this at this point.
